dandi/cli/cmd_search.py

```diff
@@ -0,0 +1,212 @@
+    for el in value:
+        value_parse += el.split(",")
+    if param.name == "select_fields":
+        if ctx.params["search_type"] == "dandisets":
+            choice_list = DANDISETS_FIELDS.keys()
+        elif ctx.params["search_type"] == "assets":
+            choice_list = ASSETS_FIELDS.keys()
+        else:
+            choice_list = None
+    else:
+        choice_list = None
+    # checking if all values are in the list of possible choices
+    for el in value_parse:
+        if choice_list and el not in choice_list:
+            ctx.fail(f"{el} is not in the list: {choice_list}")
+    return value_parse
+
+
+@click.command(help="Search TODO")
+@click.option(
+    "-F",
+    "--file",
+    help="Comma-separated list of fields to display. "
+    "An empty value to trigger a list of "
+    "available fields to be printed out",
+)
+@click.option(
+    "-t",
+    "--search_type",
+    help="Type of the search.",
+    type=click.Choice(["dandisets", "assets"]),
+)
+@click.option(
+    "-s",
+    "--select_fields",
+    help="Field name for dandisets search",
+    callback=parse_validate,
+    multiple=True,
+)
+@click.option(
+    "-f",
+    "--filter_fields",
+    help="Field name for dandisets search",
+    type=(str, str),
+    multiple=True,
+)
+@click.option(
+    "--format",
+    help="Choose the format for output. TODO",
+    type=click.Choice(["stdout", "csv"]),
+    default="stdout",
+)
+@click.option(
+    "--number_of_lines",
+    help="Number of lines of output that will be printed",
+    default=10,
+)
+@click.option(
+    "-d",
+    "--database_name",
+    help="Database name",
+    default="dandisets_new",
+)
+@map_to_click_exceptions
+def search(
+    file=None,
+    search_type=None,
+    select_fields=None,
+    filter_fields=None,
+    format="stdout",
+    number_of_lines=10,
+    database_name="dandisets_new",
+):
+
+    if file and search_type:
+        raise Exception("file and type are mutually exclusive options")
+
+    if file:
+        filepath = Path(file)
+        with filepath.open() as f:
+            query_str = f.read()
+    elif search_type in ["dandisets", "assets"]:
+        if not select_fields:
+            raise Exception(
+                f"select_fields is required if search type is {search_type}"
+            )
+        if filter_fields:
+            for el in filter_fields:
+                if el[0] not in select_fields:
+                    raise Exception(
+                        f"field {el[0]} used in filter_fields, "
+                        f"but select fields contain {select_fields}"
+                    )
+        if search_type == "dandisets":
+            query_str = create_dandisets_query(select_fields, filter_fields)
+        elif search_type == "assets":
+            query_str = create_assets_query(select_fields, filter_fields)
+    else:
+        raise NotImplementedError
+
+    endpoint = "https://search.dandiarchive.org:5820/dandisets_new/query"
```

Comment:
this would be just 1 type of search. We already have a few more (even if not exposed via API): https://llmsearch.dandiarchive.org/, https://dandiarchive.org/search etc.  

I think we should generalize this `dandi search [options] [query]` interface into `dandi search [common-options] HOW [how-specific-options-if-any] [query]` or `dandi search [options] [--how HOW] [query]` interface, so we could expand etc.